src/ccl_core.c

```diff
@@ -309,7 +345,21 @@ ccl_parameters ccl_parameters_create(
 				     double bcm_ks, int nz_mgrowth, double *zarr_mgrowth,
 				     double *dfarr_mgrowth, int *status)
 {
+  #ifndef USE_GSL_ERROR
```

Comment:
The issue is that setting up the parameters requires calculations (e.g., Omega for massive neutrinos) that depend on the GSL settings, such that the ccl_gsl object has to be populated before the create_cosmology functions are called. 

I think the underlying issue is an over-reliance on the cosmology object. For example, the current error handling procedure of writing a status message into the cosmology object doesn't cover the cases when errors happen without a cosmology being present and give raise to the awkward pair of `ccl_check_status` and `ccl_check_status_nocosmo`. But that is an issue that is outside the scope of this PR, I think.

I agree that calling it `ccl_cosmology_read_config` is now even more of a misnomer than before but I didn't want to change the API for this PR. I added a short comment before calling `ccl_cosmology_read_config` about what is going on. Which other code blocks are you referring to?

I also think that it might be worthwhile to rewrite the current setup of `ccl_splines` and `ccl_gsl` at some point in the future. The global pointers are somewhat problematic (thread- and user-safety) and personally I'm not a fan of setting environmental variables to point to config files.